lib/src/Uncertainty/Distribution/HistogramFactory.cxx

```diff
@@ -133,6 +140,63 @@ Histogram HistogramFactory::buildAsHistogram(const Sample & sample,
   return result;
 }
 
+Histogram HistogramFactory::buildAsHistogram(const Sample & sample,
+    const Scalar first,
+    const Point width) const
+{
+  const UnsignedInteger size = sample.getSize();
+  if (size == 0) throw InvalidArgumentException(HERE) << "Error: cannot build an Histogram based on an empty sample.";
+  if (sample.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: can build an Histogram only if dimension equals 1, here dimension=" << sample.getDimension();
+  const UnsignedInteger binNumber = width.getSize();
+  if (binNumber == 0) throw InvalidArgumentException(HERE) << "Error: expected a positive number of bin, got 0.";
+  // Construct the histogram
+  Point heights(binNumber, 0.0);
+  // It will extends from min to max.
+  const Scalar minvalue = sample.getMin()[0];
+  const Scalar maxvalue = sample.getMax()[0];
+  if (!SpecFunc::IsNormal(minvalue) || !SpecFunc::IsNormal(maxvalue)) throw InvalidArgumentException(HERE) << "Error: cannot build an Histogram distribution if data contains NaN or Inf";
+  if (maxvalue == minvalue)
+  {
+    const Scalar epsilon = ResourceMap::GetAsScalar("Distribution-DefaultCDFEpsilon");
+    const Scalar delta = std::max(std::abs(minvalue), 10.0) * epsilon;
+    Histogram result(minvalue - 0.5 * delta, Point(1, delta), Point(1, 1.0));
+    result.setDescription(sample.getDescription());
+    return result;
+  }
+  if (minvalue<first) throw InvalidArgumentException(HERE) << "Error: the minimum of the sample is smaller than first";
+  // Compute the right bound of the histogram = first + the sum of the widths
+  Scalar rightBound = first;
+  for(UnsignedInteger j = 0; j < binNumber; ++j)
+  {
+    rightBound += width[j];
```

Comment:
Maybe you should check that width[j] is > 0, or at least >= 0?

Reply:
That's right: OK.